modules/tinymce/src/core/test/ts/browser/api/dom/RangeExpandTest.ts

```diff
@@ -0,0 +1,91 @@
+
+describe('browser.tinymce.core.api.dom.RangeExpandTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const textEntry = 'A B ' + Unicode.nbsp + 'C D';
+
+  const makeBaseSelection = (editor: Editor, startOffset: number, endOffset: number, nodeOffset: number = 0) => {
+    const body = editor.getBody();
+    const range = editor.dom.createRng();
+    const node = body.childNodes[0].childNodes[nodeOffset];
+    range.setStart(node, startOffset);
+    range.setEnd(node, endOffset);
+    return range;
+  };
+
+  const compareRanges = (range1: Range, range2: Range) => {
+    assert.equal(range1.startContainer, range2.startContainer, 'Mismatching start container');
+    assert.equal(range1.startOffset, range2.startOffset, 'Mismatching start offset');
+    assert.equal(range1.endContainer, range2.endContainer, 'Mismatching end container');
+    assert.equal(range1.endOffset, range2.endOffset, 'Mismatching end offset');
+  };
+
+  it('TINY-9001: The content is empty', () => {
+    const editor = hook.editor();
+    editor.setContent('');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 0 ], 0);
+    const startRange = editor.selection.getRng();
+    RangeUtils(editor).expand(startRange);
+    compareRanges(startRange, editor.selection.getRng());
+  });
+
+  it('TINY-9001: The cursor is in between two spaces', () => {
+    const editor = hook.editor();
+    editor.setContent(textEntry);
+    const startRange = makeBaseSelection(editor, 4, 4);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(startRange, endRange);
+  });
+
+  it('TINY-9001: The cursor is between a space and a word on the right', () => {
+    const editor = hook.editor();
+    editor.setContent(textEntry);
+    const startRange = makeBaseSelection(editor, 5, 5);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(makeBaseSelection(editor, 5, 6), endRange);
+  });
+
+  // TODO Disabled due to inconsistent behavior between left and right, see TINY-9029
+  it.skip('TINY-9001: The cursor is between a space and a word on the left', () => {
+    const editor = hook.editor();
+    editor.setContent(textEntry);
+    const startRange = makeBaseSelection(editor, 3, 3);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(makeBaseSelection(editor, 2, 3), endRange);
+  });
+
+  it('TINY-9001: The cursor is in the middle of a word', () => {
+    const editor = hook.editor();
+    editor.setContent('<p>A B12C D</p>');
+    const startRange = makeBaseSelection(editor, 4, 4);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(makeBaseSelection(editor, 2, 6), endRange);
+  });
+
+  // TODO Disabled due to inconsistent behavior between left and right, see TINY-9029
+  it.skip('TINY-9001: The cursor is between a space and a block element on the right', () => {
```

Comment:
This one is weird though you are right it should not expand if the caret is at the beginning of the 'B' word in this case.